scalar_test.go

```diff
@@ -88,51 +88,84 @@ func TestConstructMutationID(t *testing.T) {
 	autopilot.Equals(t, `mutation MyMutation($id1:ID!$id2:ID){account{myMutation(id1: $id1 id2: $id2){id}}}`, query)
 }
 
-type IdentifierTester struct {
-	IdInputConcrete          ol.IdentifierInput  `json:"idInputConcrete"`
-	IdInputConcreteOmitEmpty ol.IdentifierInput  `json:"idInputConcreteNullable,omitempty"`
-	IdInputPointer           *ol.IdentifierInput `json:"idInputPointer"`
-	IdInputPointerOmitEmpty  *ol.IdentifierInput `json:"idInputPointerNullable,omitempty"`
+func TestMarshalIdentifiers(t *testing.T) {
+	type TestCase struct {
+		Name         string
+		Identifier   *ol.IdentifierInput
+		OutputBuffer string
+	}
+	testCases := []TestCase{
+		{
+			Name:         "the special empty identifier",
+			Identifier:   ol.EmptyIdentifier(),
+			OutputBuffer: `null`,
+		},
+		{
+			Name:         "identifier with empty arg",
+			Identifier:   ol.NewIdentifier(""),
+			OutputBuffer: `{"alias":""}`,
+		},
+		{
+			Name:         "identifier with valid ID",
+			Identifier:   ol.NewIdentifier("Z2lkOi8vb3BzbGV2ZWwvSGVsbG9Xb3JsZC8xMDEw"),
+			OutputBuffer: `{"id":"Z2lkOi8vb3BzbGV2ZWwvSGVsbG9Xb3JsZC8xMDEw"}`,
+		},
+		{
+			Name:         "identifier with valid alias",
+			Identifier:   ol.NewIdentifier("hello_world"),
+			OutputBuffer: `{"alias":"hello_world"}`,
+		},
+	}
+
+	for _, testCase := range testCases {
+		buf, err := json.Marshal(testCase.Identifier)
+		autopilot.Ok(t, err)
+		autopilot.Equals(t, testCase.OutputBuffer, string(buf))
+	}
 }
 
-func TestMarshalIdentifier(t *testing.T) {
-	// Arrange
-	id1 := ol.NewIdentifier("")
-	id2 := ol.NewIdentifier("my-service")
-	id3 := ol.NewIdentifier("Z2lkOi8vMTIzNDU2Nzg5")
-	case1 := IdentifierTester{}
-	case2 := IdentifierTester{
-		IdInputConcrete:          *id1,
-		IdInputConcreteOmitEmpty: *id1,
-		IdInputPointer:           id1,
-		IdInputPointerOmitEmpty:  id1,
-	}
-	case3 := IdentifierTester{
-		IdInputConcrete:          *id2,
-		IdInputConcreteOmitEmpty: *id2,
-		IdInputPointer:           id2,
-		IdInputPointerOmitEmpty:  id2,
-	}
-	case4 := IdentifierTester{
-		IdInputConcrete:          *id3,
-		IdInputConcreteOmitEmpty: *id3,
-		IdInputPointer:           id3,
-		IdInputPointerOmitEmpty:  id3,
+func TestMarshalIdentifiersOmitBehavior(t *testing.T) {
```

Comment:
This new test covers what is being tested in `TestMarshalIdentifier`, particularly the behaviour of `omitempty` on IdentifierInputs, so they can be used to unset fields during updates or leave fields unchanged during updates.